extension/src/experiments/data/index.ts

```diff
@@ -47,43 +47,58 @@ export class ExperimentsData extends BaseData<ExpShowOutput> {
     )
 
     void this.updateAvailableBranchesToSelect(allBranches)
-    const data: ExpShowOutput = []
 
     const { branches, currentBranch } = await this.getBranchesToShowWithCurrent(
       allBranches
     )
+    let gitLog = ''
+    const order: { branch: string; sha: string }[] = []
+    const args: Args = []
 
-    await Promise.all(
-      branches.map(async branch => {
-        const branchFlags = [
-          ExperimentFlag.REV,
-          branch,
-          ExperimentFlag.NUM_COMMIT,
-          this.experiments.getNbOfCommitsToShow(branch).toString()
-        ]
-
-        const output = (await this.expShow(
-          branchFlags,
-          branch
-        )) as ExpShowOutput
-
-        if (branch !== currentBranch) {
-          const workspaceIndex = output.findIndex(
-            exp => exp.rev === EXPERIMENT_WORKSPACE_ID
-          )
-          output.splice(workspaceIndex, 1)
-        }
-        data.push(...output)
-      })
+    for (const branch of branches) {
+      gitLog = await this.collectGitLogAndOrder(gitLog, branch, order, args)
+    }
+
+    const expShow = await this.internalCommands.executeCommand<ExpShowOutput>(
+      AvailableCommands.EXP_SHOW,
+      this.dvcRoot,
+      ...args
     )
 
-    this.collectFiles(data)
+    this.collectFiles({ expShow })
 
-    return this.notifyChanged(data)
+    return this.notifyChanged({ currentBranch, expShow, gitLog, order })
   }
 
-  protected collectFiles(data: ExpShowOutput) {
-    this.collectedFiles = collectFiles(data, this.collectedFiles)
+  protected collectFiles({ expShow }: { expShow: ExpShowOutput }) {
+    this.collectedFiles = collectFiles(expShow, this.collectedFiles)
+  }
+
+  private async collectGitLogAndOrder(
+    gitLog: string,
+    branch: string,
+    order: { branch: string; sha: string }[],
+    args: Args
+  ) {
+    const nbOfCommitsToShow = this.experiments.getNbOfCommitsToShow(branch)
+
+    const branchGitLog = await this.internalCommands.executeCommand(
+      AvailableCommands.GIT_GET_COMMIT_MESSAGES,
```

Comment:
[F] As a follow up I think we want to pull the timestamp from the git log so that commits are displayed in chronological order in the tree.

Reply:
already done in here